packages/server/assets/core/typedefs/projects.graphql

```diff
@@ -0,0 +1,255 @@
+extend type Query {
+  """
+  Get a specific project.
+  """
+  project(id: String!): Project
+  """
+  Get all the projects you have access to.
+  """
+  projects(limit: Int! = 20, cursor: String, query: ProjectQuery): ProjectCollection
+  # discoverableProjects(limit: Int! = 20, cursor: String): ProjectCollection # TODO, stage 2
+}
+
+"""
+Formerly known as 'stream'.
+"""
+type Project {
+  id: String!
+  name: String!
+  description: String
+  """
+  Whether the project can be viewed by non-contributors
+  """
+  isPublic: Boolean!
+  """
+  Whether the project (if public) can be found on public project exploration pages and searches
+  """
+  isDiscoverable: Boolean!
+  """
+  Whether this project allows comments from any logged in user.
+  """
+  allowPublicComments: Boolean!
+  """
+  Your role for this project. `null` if request is not authenticated, or the project is not explicitly shared with you.
+  """
+  role: String
+  createdAt: DateTime!
+  updatedAt: DateTime!
+  """
+  The people that have access to this project.
+  """
+  team: [LimitedUser!]!
+  """
+  Collaborators who have been invited, but not yet accepted.
+  """
+  pendingTeamMemembers: [PendingProjectTeamMember!] @hasStreamRole(role: STREAM_OWNER)
+  size: String
+  """
+  Date when you favorited this project. `null` if project isn't viewed from a specific user's perspective or if it isn't favorited.
+  """
+  favoritedDate: DateTime
+  """
+  How many times this project has been favorited
+  """
+  favoritesCount: Int!
+}
+
+type ProjectCollection {
+  totalCount: Int!
+  cursor: String
+  items: [Project!]
+}
+
+input ProjectQuery {
+  """
+  Filters by ownership type or favourites.
+  """
+  filter: ProjectFilter = all
+  """
+  Searches by a specific project name.
+  """
+  nameQuery: String
+}
+
+enum ProjectFilter {
+  all
+  owner
+  contributor
+  reviewer
+  favourite
+}
+
+type PendingProjectTeamMember {
+  id: String!
+  inviteId: String!
+  projectId: String!
+  projectName: String!
+  """
+  E-mail address or name of the invited user
+  """
+  title: String!
+  role: String!
+  invitedBy: LimitedUser!
+  """
+  Set only if user is registered
+  """
+  user: LimitedUser
+  """
+  Only available if the active user is the pending project collaborator
+  """
+  token: String
+}
+
+extend type LimitedUser {
+  """
+  Returns all discoverable projects that the user is a collaborator on
+  """
+  projects(limit: Int! = 20, cursor: String): ProjectCollection!
+    @hasRole(role: "server:user")
+    @hasScope(scope: "streams:read")
+
+  """
+  Total amount of favorites attached to projects owned by the user
+  """
+  totalOwnedProjectsFavorites: Int!
+}
+
+extend type Mutation {
+  """
+  Creates a new project.
+  """
+  projectCreate(project: ProjectCreateInput!): String
+    @hasRole(role: "server:user")
```

Comment:
Just an idea - I've seen it suggested on an Apollo live stream that to organize mutations better you can nest them in group objects similarly how you would nest query fields.

So, for example, instead of having `projectCreate`, `projectDelete`, `projectUpdatePermission` as top level operations you'd have a `ProjectMutations` type and a top-level operation `projectMutations: ProjectMutations!`. `ProjectMutations` then would have fields for all of the actual operations like `create`, `update`, `delete` etc.

I've never used this kind of pattern, but superficially it does make sense as a way to make the API less cluttered and easier to understand, so I'm wondering if we shouldn't try it out